src/util/fields.js

```diff
@@ -124,75 +129,32 @@ export const COMMENTS_FIELDS_TO_FETCH = [
-    return BASE_FIELDS_TO_FETCH.concat(SIDEBAR_FIELDS_TO_FETCH).concat(PREVIEW_FIELDS_TO_FETCH);
-}
-
-/**
- * Returns base fields and preview fields
+ * Finds fields missing in box file
  *
- * @return {Array<string>} list of fields with preview
- */
-function getFieldsIncludingPreview(): Array<string> {
-    return BASE_FIELDS_TO_FETCH.concat(PREVIEW_FIELDS_TO_FETCH);
-}
-
-/**
- * Returns fields needed for fetching
- *
- * @param {boolean|void} [includePreview] - Optionally include preview fields
- * @param {boolean|void} [includePreviewSidebar] - Optionally include preview and sidebar fields
- * @return {Array<string>} list of fields
- */
-function getFields(includePreview?: boolean = false, includePreviewSidebar?: boolean = false): Array<string> {
-    let fields = BASE_FIELDS_TO_FETCH;
-    if (includePreview && includePreviewSidebar) {
-        // Only include sidebar fields if we are also including preview fields
-        fields = getFieldsIncludingPreviewSidebar();
-    } else if (includePreview) {
-        // Preview may not have a sidebar
-        fields = getFieldsIncludingPreview();
-    }
-
-    return fields;
-}
-
-/**
- * Returns fields needed for fetching
- *
- * @param {boolean|void} [includePreview] - Optionally include preview fields
- * @param {boolean|void} [includePreviewSidebar] - Optionally include preview and sidebar fields
+ * @param {boolean|void} [options.includePreviewFields] - Optionally include preview fields
+ * @param {boolean|void} [options.includeSidebarFields] - Optionally include preview and sidebar fields
  * @return {string} comma seperated list of fields
  */
-export function getFieldsAsString(includePreview?: boolean = false, includePreviewSidebar?: boolean = false): string {
-    const fields = getFields(includePreview, includePreviewSidebar);
-    return fields.join(',');
-}
-
-/**
- * Checks the fields needed for a box file
- *
- * @param {boolean|void} [includePreview] - Optionally include preview fields
- * @param {boolean|void} [includePreviewSidebar] - Optionally include preview and sidebar fields
- * @return {string} comma seperated list of fields
- */
-export function isValidBoxFile(
-    file?: BoxItem | string,
-    includePreview?: boolean = false,
-    includePreviewSidebar?: boolean = false
-): boolean {
-    if (!file || typeof file !== 'object') {
-        return false;
+function findMissingFields(file?: BoxItem, fields?: Array<string> = []): Array<string> {
+    if (!file || typeof file !== 'object' || Object.keys(file).length === 0) {
+        return fields;
     }
 
-    const fields = getFields(includePreview, includePreviewSidebar).filter(
-        (field) => BOX_ITEM_OPTIONAL_FIELDS.indexOf(field) < 0
-    );
-
-    // Some fields like metadata have dots in it. Only use the root prop.
-    return fields.every((field) => typeof getProp(file, field.split('.')[0]) !== 'undefined');
+    return fields.filter((field: string) => {
```

Comment:
Why not rely on the whole string? We may end up at a point where different components want different metadata fields.

Reply:
Because when you make a request for ?fields=foo.bar.baz, the API will return `{ foo: null }` suggesting there is no `foo` and not `{ foo: { bar: null } }` suggesting there is no `baz`. So if I use the full path, then it will always consider `foo.bar.baz` to be missing. Only solution is that I need to check `foo.bar.baz` and then `foo.bar` and then `foo` to correctly figure out the missing field.

Reply:
Looks pretty ugly/hacky

```js
function findMissingFields(file, fields) {
    return fields.filter((field) => {
        let hasField = _.has(file, field);
        let hasParent = false;

        while (field && !hasField) {
            field = field.substring(0, field.lastIndexOf('.'));
            hasParent = _.has(file, field)
        }

        return !hasField || hasParent;
    });
}
```